lib/protocol/bep_extensions.go

```diff
@@ -394,3 +395,20 @@ func VectorHash(v Vector) []byte {
 	}
 	return h.Sum(nil)
 }
+
+func (x *FileInfoType) MarshalJSON() ([]byte, error) {
+	return json.Marshal(x.String())
+}
+
+func (x *FileInfoType) UnmarshalJSON(data []byte) error {
+	var s string
+	if err := json.Unmarshal(data, &s); err != nil {
+		return err
+	}
+	n, ok := FileInfoType_value[s]
+	if !ok {
+		return errors.New("invalid value: " + s)
+	}
+	*x = FileInfoType(n)
+	return nil
+}
```

Comment:
This adds some more potential to break something, e.g. a wrapper that shows out-of-sync files with a file type indication. Not very likely either though.

And this looks so generic I'd expect some `asstring` proto option - turns out doesn't exist because jsonpb does it by default.